forc-plugins/forc-doc/src/render/title.rs

```diff
@@ -0,0 +1,94 @@
+use sway_core::language::ty::TyDecl;
+
+pub(crate) trait DocBlockTitle {
+    fn as_block_title(&self) -> BlockTitle;
+}
+/// Represents all of the possible titles
+/// belonging to an index or sidebar.
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq)]
+pub(crate) enum BlockTitle {
+    Modules,
+    Structs,
+    Enums,
+    Traits,
+    Abi,
+    ContractStorage,
+    Constants,
+    Functions,
+    Fields,
+    Variants,
+    RequiredMethods,
+}
+impl BlockTitle {
+    pub(crate) fn as_str(&self) -> &str {
+        match self {
+            Self::Modules => "Modules",
+            Self::Structs => "Structs",
+            Self::Enums => "Enums",
+            Self::Traits => "Traits",
+            Self::Abi => "Abi",
+            Self::ContractStorage => "Contract Storage",
+            Self::Constants => "Constants",
+            Self::Functions => "Functions",
+            Self::Fields => "Fields",
+            Self::Variants => "Variants",
+            Self::RequiredMethods => "Required Methods",
+        }
+    }
+    pub(crate) fn item_title_str(&self) -> &str {
+        match self {
+            Self::Modules => "Module",
+            Self::Structs => "Struct",
+            Self::Enums => "Enum",
+            Self::Traits => "Trait",
+            Self::Abi => "Abi",
+            Self::ContractStorage => "Contract Storage",
+            Self::Constants => "Constant",
+            Self::Functions => "Function",
+            Self::Fields => "Fields",
+            Self::Variants => "Variants",
+            Self::RequiredMethods => "Required Methods",
+        }
+    }
+    pub(crate) fn class_title_str(&self) -> &str {
+        match self {
+            Self::Modules => "mod",
+            Self::Structs => "struct",
+            Self::Enums => "enum",
+            Self::Traits => "trait",
+            Self::Abi => "abi",
+            Self::ContractStorage => "storage",
+            Self::Constants => "constant",
+            Self::Functions => "fn",
+            _ => unimplemented!("These titles are unimplemented, and should not be used this way."),
```

Comment:
Shouldn't this use a `panic!` instead? Wouldn't that be more accurate since it's not meant to be implemented?

Reply:
Since it's only for use in forc doc, I think it makes sense to have them as unimplemented or unreachable. cc @JoshuaBatty

Reply:
I'm happy with it staying as `unimplemented!`